cluster/saltbase/salt/kube-apiserver/default

```diff
@@ -4,10 +4,15 @@
 {% endif -%}
 
 {% set cloud_provider = "" -%}
-{% if grains.cloud_provider is defined -%}
-  {% set cloud_provider = "--cloud_provider=" + grains.cloud_provider -%}
+{% if grains.cloud is defined -%}
+{% set cloud_provider = "--cloud_provider=" + grains.cloud -%}
+
+{% if grains.cloud == 'aws' -%}
+  {% set cloud_config = "--cloud_config=/etc/aws.conf" -%}
```

Comment:
line 38 rewrites cloud config to empty string...

you need to move that line above this line so it has an initially empty value and is not reset later in file.